servlet-transport/src/main/java/fi/nls/oskari/transport/TransportService.java

```diff
@@ -719,6 +724,32 @@ private void setFilter(SessionStore store, String json, Map<String, Object> para
         }
     }
     /**
+     * Writes features to file in requested export format
+     * - will be request just once.
+     *
+     * .
+     *
+     * @param store
+     * @param json
+     */
+    private void setExport(SessionStore store, String json, Map<String, Object> params) {
+
+        String format = params.get(PARAM_FORMAT).toString();
+        // stores format, but doesn't save
+        store.setExportFormat(format);
+        String layerId = params.get(PARAM_LAYER_ID).toString();
+
+        if (store.containsLayer(layerId)) {
+            // TODO: Check export/load permission
+            Job job = createOWSMapLayerJob(createResultProcessor(parseRequestId(params)),
+                    JobType.EXPORT, store, layerId, false, true, false, false);
+            if (job != null) {
+                jobs.add(job);
+            }
+
+        }
+    }
```

Comment:
This should probably throw an exception or otherwise signal the frontend that we're not actually going to do anything about the request (in else branch):
            throw new ServiceRuntimeException("Layer is not selected",
                    WFSExceptionHelper.ERROR_SET_PROCESS_REQUEST_FAILED, WFSExceptionHelper.WARNING_LEVEL);